### PR TITLE
Docs: Added pushing properties to Readme

### LINKED ISSUE #212: Documentation: Pushing properties to logevents
This integration is great, it enables developers to just use the default ILogger in dotnet and configure the logging with serilog. I was however missing the following information, as found [here](https://nblumhardt.com/2016/11/ilogger-beginscope/)

It would be great to have this added to the main readme of this module.

```plain
1. If the state argument is IEnumerable<KeyValuePair<string, object>>, the values are attached to the event directly
This may not be entirely intuitive, if you think of Scope as a trail of breadcrumbs. If you’ve dealt with correlating log events in a distributed application, you’ll recognize that often you’re interested in log events related to a specific transaction, customer, request, thread, machine or endpoint.

In the absence of any other API for the purpose, BeginScope() is how these correlation ids need to be attached in Microsoft.Extensions.Logging:

using (logger.BeginScope(new Dictionary<string, object>{
    ["CustomerId"] = 12345,
    ["OrderId"] = 54
}))
{
    logger.LogInformation("Processing credit card payment");
}
In this case, events created in the scope will have two additional properties attached: CustomerId and OrderId.

Types that implement IEnumerable<KeyValuePair<string, object>> rarely have any other useful representation; ToString() on a dictionary generally just provides the type name. These values therefore don’t result in a breadcrumb-like Scope property.
```

Something like:

## Pushing properties to the logger

When using the `ILogger` from **Microsoft.Extensions.Logging** you cannot push properties like you would with the ILogger from Serilog. If you still want to push properties you can do with the following code:

```csharp
using (logger.BeginScope(new Dictionary<string, object>
            {
                ["UserId"] = "xxx",
                ["ExtraProperty"] = "yyy",
            }))
{
   // UserId and ExtraProperty are set for all logging events in these brackets
}
```

The above code results in the same as if using the regular serilog ILogger with this code

```csharp
using (logger.PushProperty("UserId", "xxx"))
using (logger.PushProperty("ExtraProperty", "yyy"))
{
    // UserId and ExtraProperty are set for all logging events in these brackets
}
```

### PR DESCRIPTION
**Pull request by svrooij**
Fixed #212

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -275,3 +275,31 @@ The Azure Diagnostic Log Stream ships events from any files in the `D:\home\LogF
...
+{
+    ["UserId"] = "xxx",
+    ["ExtraProperty"] = "yyy",
+}))
+{
+   // UserId and ExtraProperty are set for all logging events in these brackets
+}
+```
+
+The code above results in the same outcome as if you would push propeties in the **ILogger** in Serilog.
```
Comment (sungam3r) [inline README.md:296]:
```suggestion
The code above results in the same outcome as if you would push properties in the **ILogger** in Serilog.
```

File: README.md [hunk 2]
```diff
@@ -275,3 +275,31 @@ The Azure Diagnostic Log Stream ships events from any files in the `D:\home\LogF
                 flushToDiskInterval: TimeSpan.FromSeconds(1))
             .CreateLogger();
 ```
+
+### Pushing properties to the ILogger
```
Comment (nblumhardt) [inline README.md:279]: Nit: maybe:

```md
### Pushing properties to `ILogger<T>`
```

?

File: README.md [hunk 3]
```diff
@@ -275,3 +275,31 @@ The Azure Diagnostic Log Stream ships events from any files in the `D:\home\LogF
...
+```csharp
+// Microsoft.Extensions.Logging ILogger
+// Yes, it's required to use a dictionary. See https://nblumhardt.com/2016/11/ilogger-beginscope/
+using (logger.BeginScope(new Dictionary<string, object>
+{
+    ["UserId"] = "xxx",
+    ["ExtraProperty"] = "yyy",
+}))
+{
+   // UserId and ExtraProperty are set for all logging events in these brackets
```
Comment (nblumhardt) [inline README.md:292]: The README currently avoids `foo`/`bar`/`xxx` etc., could we try something like:

```csharp
    ["UserId"] = "user-123",
    ["OperationType"] = "update",
```

or something similar?

**Review by nblumhardt — COMMENTED** [review]
Just some nitpicky suggestions from me, looks good!

File: README.md (hunk 2 above)
Comment (svrooij) [reply README.md]: Yes why not, but github didn’t like my formatting, so I converted it to code with ``` around it.

File: README.md (hunk 3 above)
Comment (svrooij) [reply README.md]: I changed the properties to something more reasonable.

**Comment by svrooij** [comment]
I’m ok with squashing this PR, because it looks like a mess right now. For commits to change one alinea in the readme. Just squash and add this to the commit message:

`Co-authored-by: Stephan van Rooij <1292510+svrooij@users.noreply.github.com> `

**Comment by nblumhardt** [comment]
Thanks 👍

**Comment by svrooij** [comment]
You're welcome

**Pull request merged**